### PR TITLE
Ovirt move network to cluster

### LINKED ISSUE #33852: Move setting cluster network roles from ovirt_networks module to ovirt_cluster module?
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

modules: ovirt_networks, ovirt_cluster

##### ANSIBLE VERSION

```
ansible 2.4.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/djasa/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.14 (default, Nov  3 2017, 10:55:25) [GCC 7.2.1 20170915 (Red Hat 7.2.1-2)]
```

##### OS / ENVIRONMENT

N/A

##### SUMMARY
Cluster network roles would be better set up by ovirt_cluster module to avoid duplication and errors

##### LONG DESCRIPTION
when typing my existing RHV setup to ansible, I was wondering why are cluster network roles specified at each network instead of at cluster. The main disadvantages are IMO:
* much more typing. E.g. you have to specify `display: false` for each network instead of having single `display: NETWORK_NAME` statement given at cluster
* vulnerability to errors for no gain: with per-cluster `display: NETWORK_NAME`, you can validate that just one network is specified as display right in the ansible module (the same is true for migration network)

I don't have a strong opinion for VM networks and required networks specification where the inherent validation and less typing are not really applicable (it doesn't matter that much if you specify per-network boolean or per-cluster list in this case).

The only downside I can think of is need to have cluster set up split to two tasks but that's IMO an acceptable cost, see example below:
```yaml
- name: set up clusters
  ovirt_cluster:
    # everything but networks

- name: set up networks
  ovirt_networks:
    # now set up networks

- name: set up cluster network roles
  ovirt_cluster:
    name: cluster_name
    display_network: my_display_network
    migration_network: my_mgmt_network
    # These are of no real benefit IMO so just for consistency sake...
    required_networks:
    - my_mgmt_network
    - my_display_network
    - infra_servers
    - prod_servers
    vm_networks:
    - infra_servers
    - web_servers
    - playground
```

(originally misfiled as https://github.com/oVirt/ovirt-ansible/issues/142)

### PR DESCRIPTION
**Pull request by mnecas**
##### SUMMARY

Ovirt move network to cluster

Fixes #33852
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

ovirt
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/ovirt/ovirt_cluster.py [hunk 1]
```diff
@@ -265,6 +265,25 @@
             - "Could be for example I(virtual-host), I(rhgs-sequential-io), I(rhgs-random-io)"
         version_added: 2.8
         type: str
+    networks:
+        description:
+            - List of cluster networks.
+            - You can only create or delete them for editing use ovirt_network module.
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_cluster.py:271]: M(ovirt_network)

Comment (machacekondra) [reply lib/ansible/modules/cloud/ovirt/ovirt_cluster.py:271]: Please don't create/delete the network, just fail if it don't exist

File: lib/ansible/modules/cloud/ovirt/ovirt_cluster.py [hunk 2]
```diff
@@ -265,6 +265,25 @@
...
+        suboptions:
+            name:
+                description:
+                    - Name of the network cluster.
+            type:
+                description:
+                    - This type indicates the purpose that the network is used for in the cluster.
+                choices: ['display', 'gluster', 'migration', 'management', 'default_route', 'vm']
+            state:
```
Comment (mnecas) [inline lib/ansible/modules/cloud/ovirt/ovirt_cluster.py:280]: assigned

**Comment by mnecas** [comment]
I'm closing this because it is not essential to fix, it can be done from multiple sides.
If needed we can continue on this PR in ovirt collection.
https://github.com/oVirt/ovirt-ansible-collection

**Pull request closed without merging**